src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -635,8 +636,15 @@ public async Task PublishZipDeploy(Site functionApp, Func<Task<Stream>> zipFileF
 
         public async Task<DeployStatus> PerformServerSideBuild(Site functionApp, Func<Task<Stream>> zipFileFactory, Func<HttpClient, Task<DeployStatus>> deploymentStatusPollTask, bool attachRestrictedToken = true)
         {
+            Uri scmUri = null;
+            if (!Uri.TryCreate($"https://{functionApp.ScmUri}", UriKind.Absolute, out scmUri))
+            {
+                throw new CliException($"Your function app {functionApp.SiteName} does not support remote build. " + 
+                    "To enable remote build, please update your function app to the latest verison by recreating it.");
+            }
```

Comment:
Is the only way to recreate it? Or can they use AppSettings update or something?

Reply:
We checked that, the app settings cannot change function app metadata in datarole. Sadly.